spot_driver/include/spot_driver/api/middleware_handle_base.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2024 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <rclcpp/qos.hpp>
+
+namespace spot_ros2 {
+
+/**
+ * @brief Base class for common ROS2 publisher definitions
+ */
+class MiddlewareHandleBase {
+ public:
+  rclcpp::QoS makePublisherQoS(size_t const publisherHistoryDepth) const {
+    // most compatible publisher durability: transient local
+    // most compatible publisher reliabilty: reliable
+    // see also
+    // https://docs.ros.org/en/iron/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-compatibilities
+    return rclcpp::QoS(publisherHistoryDepth).transient_local().reliable();
```

Comment:
@bhung-bdai @khughes-bdai did you guys change the QoS to best effort? Just realized I forgot to do that before testing again lol and the errors returned

Reply:
I get a bunch of mismatched QoS from the nodes we generate withe `image_pipeline`'s `depth_proc_node`s. I had looked into their source code with Mich last time and saw they set their QoS's all to ROS default (reliable)

```
[component_container_mt-8] [WARN] [1715004057.643811002] [Opal.register_node_right]: New publisher discovered on topic '/Opal/depth/right/image', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.644263948] [Opal.register_node_right]: New publisher discovered on topic '/Opal/depth/right/camera_info', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.644683021] [Opal.point_cloud_xyzrgb_node_hand]: New publisher discovered on topic '/Opal/camera/hand/image', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.644960723] [Opal.point_cloud_xyzrgb_node_hand]: New publisher discovered on topic '/Opal/camera/hand/camera_info', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.644978954] [Opal.register_node_hand]: New publisher discovered on topic '/Opal/camera/hand/camera_info', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.645372683] [Opal.register_node_hand]: New publisher discovered on topic '/Opal/depth/hand/image', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[component_container_mt-8] [WARN] [1715004057.645755978] [Opal.register_node_hand]: New publisher discovered on topic '/Opal/depth/hand/camera_info', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646131217] [Opal.image_publisher]: New subscription discovered on topic '/Opal/camera/back/image', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646303530] [Opal.image_publisher]: New subscription discovered on topic '/Opal/camera/back/camera_info', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646323736] [Opal.image_publisher]: New subscription discovered on topic '/Opal/depth/back/image', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646337799] [Opal.image_publisher]: New subscription discovered on topic '/Opal/depth/back/camera_info', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646349650] [Opal.image_publisher]: New subscription discovered on topic '/Opal/camera/frontleft/image', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
[spot_image_publisher_node-2] [WARN] [1715004057.646361215] [Opal.image_publisher]: New subscription discovered on topic '/Opal/camera/frontleft/camera_info', requesting incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY
```